releases/latest/full/Dockerfile.ubi.openjdk21

```diff
@@ -1,155 +1,43 @@
-
-COPY NOTICES /opt/ol/NOTICES
-COPY helpers /opt/ol/helpers
-COPY fixes/ /opt/ol/fixes/
-
-# Install dumb-init
+# Install all features
 RUN set -eux; \
-    ARCH="$(uname -m)"; \
-    case "${ARCH}" in \
-       aarch64|arm64) \
-         DUMB_INIT_URL='https://github.com/Yelp/dumb-init/releases/download/v1.2.5/dumb-init_1.2.5_aarch64'; \
-         DUMB_INIT_SHA256=b7d648f97154a99c539b63c55979cd29f005f88430fb383007fe3458340b795e; \
-         ;; \
-       amd64|x86_64) \
-         DUMB_INIT_URL='https://github.com/Yelp/dumb-init/releases/download/v1.2.5/dumb-init_1.2.5_x86_64'; \
-         DUMB_INIT_SHA256=e874b55f3279ca41415d290c512a7ba9d08f98041b28ae7c2acb19a545f1c4df; \
-         ;; \
-       ppc64el|ppc64le) \
-         DUMB_INIT_URL='https://github.com/Yelp/dumb-init/releases/download/v1.2.5/dumb-init_1.2.5_ppc64le'; \
-         DUMB_INIT_SHA256=3d15e80e29f0f4fa1fc686b00613a2220bc37e83a35283d4b4cca1fbd0a5609f; \
-         ;; \
-       s390x) \
-         DUMB_INIT_URL='https://github.com/Yelp/dumb-init/releases/download/v1.2.5/dumb-init_1.2.5_s390x'; \
-         DUMB_INIT_SHA256=47e4601b152fc6dcb1891e66c30ecc62a2939fd7ffd1515a7c30f281cfec53b7; \
-         ;;\
-       *) \
-         echo "Unsupported arch: ${ARCH}"; \
-         exit 1; \
-         ;; \
-    esac; \
-    curl -LfsSo /usr/bin/dumb-init ${DUMB_INIT_URL}; \
-    echo "${DUMB_INIT_SHA256} */usr/bin/dumb-init" | sha256sum -c -; \
-    chmod +x /usr/bin/dumb-init;
-
-# Add default user 1001 and create wlp with right user/permissions before copying
-RUN microdnf -y install shadow-utils findutils openssl \
-    && adduser -u 1001 -r -g root -s /usr/sbin/nologin default \
-    && mkdir -p /opt/ol/wlp \
-    && chown -R 1001:0 /opt/ol/wlp \
-    && chmod -R g+rw /opt/ol/wlp \
-    && microdnf remove -y shadow-utils \
-    && microdnf clean all
-
-# Copy the runtime and licenses
-COPY --from=getRuntime --chown=1001:0 /opt/ol/wlp /opt/ol/wlp
-COPY --from=getRuntime /licenses /licenses
-
-
-# Set Path Shortcuts
-ENV PATH=$PATH:/opt/ol/wlp/bin:/opt/ol/helpers/build \
-    LOG_DIR=/logs \
-    WLP_OUTPUT_DIR=/opt/ol/wlp/output \
-    WLP_SKIP_MAXPERMSIZE=true \
-    OPENJ9_SCC=$OPENJ9_SCC
+  if [ ! -f /tmp/openliberty-MavenArtifact-$LIBERTY_VERSION.zip ]; then \
+    curl https://repo1.maven.org/maven2/io/openliberty/features/features/$LIBERTY_VERSION/features-$LIBERTY_VERSION.json > /tmp/features-$LIBERTY_VERSION.json; \
+  else \
+    mkdir /tmp/feature-repo-$LIBERTY_VERSION; \
+    unzip -d /tmp/feature-repo-$LIBERTY_VERSION /tmp/openliberty-MavenArtifact-$LIBERTY_VERSION.zip; \
+    cp /tmp/feature-repo-$LIBERTY_VERSION/io/openliberty/features/features/$LIBERTY_VERSION/features-$LIBERTY_VERSION.json /tmp; \
+    export FEATURE_LOCAL_REPO=/tmp/feature-repo-$LIBERTY_VERSION; \
+    export FEATURE_VERIFY=skip; \
+  fi; \
```

Comment:
Should there be some verification of the SHA of the repo zip here, similar to the way we did the `wlp.zip`.

Regardless of the need to verify the SHA of the repo zip.  I do think a SHA verification is need for the `.json` file in both cases of it being downloaded from maven or extracted out of the existing repo zip.

Reply:
I agree with adding SHA validation. 

My suggestion would be to perform the SHA validation only on the maven artifact zip (which includes the features json file) in the pre-release scenario and to perform the SHA validation on features json file in the production/GA build scenario.

The same build argument could be used in both cases:
`ARG LIBERTY_FEATURES_SHA={replace_with_correct_features_sha}`

The value will be set in the Dockerfile once the features json is published to maven (at GA). For pre-release, the build script can set the same build argument with the SHA value of the zip file (similar to wlp.zip).

FYI @wraschke - `create-new-release.sh` will need to be updated I believe to account for the above.